src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -104,10 +105,23 @@ public IEnumerable<string> SupportedVersions
         /// <inheritdoc/>
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
-            var detectionResult = _detector.Detect(new DetectorContext
+            PlatformDetectorResult detectionResult;
+
+            try
             {
-                SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
-            });
+                detectionResult = _detector.Detect(new DetectorContext
+                {
+                    SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
+                });  
+            }
+            catch (FailedToParseFileException ex)
```

Comment:
Shouldn't we need this guard for all other platforms too since any detector could throw this exception? That said I wonder if we should have a generic exception that a Detector throws and that we can catch in all platforms? I say this because from a platform's perspective this is an internal detail of the detector and there is a possibility of other kind of exceptions that could be thrown from a detector which also could be interesting for a user to react (ex: fix my toml file)

Reply:
We should add unit tests for each platform to see if we handling this properly. for example, mocking the `IPythonPlatformDetector` to throw and see how `PythonPlatform` reacts

Reply:
I agree we should have a generic exception for detector only and will work on adding more tests, thanks!